C/Cpp_include/c4ConnectedClient.hh

```diff
@@ -0,0 +1,37 @@
+//
+// c4ConnectedClient.hh
+//
+// Copyright 2021-Present Couchbase, Inc.
+//
+// Use of this software is governed by the Business Source License included
+// in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+// in that file, in accordance with the Business Source License, use of this
+// software will be governed by the Apache License, Version 2.0, included in
+// the file licenses/APL2.txt.
+//
+
+#pragma once
+#include "c4Base.hh"
+#include "Async.hh"
```

Comment:
This is interesting ... I hadn't considered exposing Async as part of the LiteCore C++ API; I considered it internal like Actor. But it would definitely simplify this API. And Async.hh has few dependencies. I'll think about this.